sdk/core/azure_core/src/http/pager.rs

```diff
@@ -408,11 +423,73 @@ impl<P: Page> ItemIterator<P> {
     /// You can use this to asynchronously iterate pages returned by a collection request to a service.
     /// This allows you to get the individual pages' [`Response<P>`], from which you can iterate items in each page
     /// or deserialize the raw response as appropriate.
+    ///
+    /// The returned `PageIterator` resumes from the current page until _after_ all items are processed.
+    /// It does not continue on the next page until you call `next()` after the last item in the current page
+    /// because of how iterators are implemented. This may yield duplicates but will reduce the likelihood of skipping items instead.
     pub fn into_pages(self) -> PageIterator<P> {
+        // Attempt to start paging from the current page so that we don't skip items,
+        // assuming the service collection hasn't changed (most services don't create ephemeral snapshots).
+        if let Ok(mut token) = self.next_token.lock() {
+            *token = self.continuation_token;
+        }
+
         PageIterator {
             stream: self.stream,
-            continuation_token: Default::default(),
+            continuation_token: self.next_token,
+        }
+    }
+
+    /// Advance the `ItemIterator` to the page referenced by `continuation_token`.
+    ///
+    /// You should call this before iterating the [`Stream`] or results may be unpredictable.
```

Comment:
Do you want to specifically say `ItemIterator` here rather than `Stream`? I'm not sure it really matters though.